handler/src/main/java/io/netty/handler/ssl/util/SimpleTrustManagerFactory.java

```diff
@@ -128,7 +132,20 @@ protected void engineInit(
 
         @Override
         protected TrustManager[] engineGetTrustManagers() {
-            return parent.engineGetTrustManagers();
+            TrustManager[] trustManagers = this.trustManagers;
```

Comment:
Why? `this.trustManagers` is volatile. Any changed made before writing to `this.trustManagers` (`this.trustManagers = trustManagers`) should be visible after reading `this.trustManagers` (`trustManagers = this.trustManagers`). Also, it's OK to cache `trustManagers` twice because it's just wrapping.